sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobOutputStream.java

```diff
@@ -114,6 +132,10 @@ static BlobOutputStream pageBlobOutputStream(final PageBlobAsyncClient client, f
     @Override
     public synchronized void close() throws IOException {
         try {
+            // check if multiple closes are expected to be made and stream is already closed
+            if (isMultipleClosedAllowed && !isAlreadyClosed.compareAndSet(false, true)) {
+                return;
+            }
```

Comment:
If the first `close()` attempt tried to `commit` and failed with a retriable exception (like timeout), the previous behavior allowed the user to call `close()` again to attempt to commit. Since that is no longer possible, we should ensure the exception message is updated on line no. 150 to specify that the user will have to `flush` and `commit` manually to create the blob. 

The other option is to allow closing again only if the first `close()`  did not succeed but this will require maintaining additional state and knowing which exception is retriable. If the service throws an exception because the blob was already created and cannot be overwritten, then we know the first attempt succeeded on the service side and we just ignore the "resource already exists exception" and consider `close()` as successful. All subsequent `close()` calls will return immediately.

Reply:
The retry policy should be already taking care of retriable exceptions.
The "resource already exists" is kind of grey area, it might happen if concurrent transaction finished first.

Perhaps we should just.
```
if (alreadySuccessfullyClosed) {
  // do nothing
} else {
  // existing logic.
}
```

Reply:
> The "resource already exists" is kind of grey area, it might also happen if concurrent transaction finished first.

Either way, the `commit` didn't complete successfully and we get the exception to indicate that the blob already exists. So, we don't attempt to commit again and consider it to be "alreadySuccessfullyClosed" when close is called again. Basically, do what you are suggesting above with the additional consideration that "blob already exists" (whether it's from this commit or from another concurrent transaction) will be treated as a success.